### PR TITLE
FS-1053 Fix firmware record updates for existing firmware

### PR DESCRIPTION
**Pull request by coffeefreak101**
The firmware-syncer is not updating records in serverservice for firmware with existing files in the S3 bucket.

This change refactors the vendor code to reduce some code duplication and fix the issue in one place instead of several.

Also refactored some of the code to make it more testable, and added some mocks.

### CODE REVIEW

File: internal/vendors/equinix/equinix.go [hunk 1]
```diff
@@ -5,159 +5,80 @@ import (
...
 	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
+
 	serverservice "go.hollow.sh/serverservice/pkg/api/v1"
 )
 
 const GithubDownloadTimeout = 300
 
-// Equinix implements the Vendor interface methods to retrieve Equinix OpenBMC firmware files
-type Equinix struct {
```
Comment (diogomatsubara) [inline internal/vendors/equinix/equinix.go:27]: Now that the equinix vendor is gone, it'd make sense to change the package name and the filename to `github` and `github/github.go` respectively

**Review by diogomatsubara — APPROVED** [review]
These changes look great! It's really good to have another set of eyes on this codebase and improve it as you did. Thank you @coffeefreak101 ! I made one comment about changing the `equinix` package to `github`. It'd be nice to also deal with all the lint warnings. Other than that, LGTM

File: internal/vendors/equinix/equinix.go (hunk 1 above)
Comment (coffeefreak101) [reply internal/vendors/equinix/equinix.go:27]: That makes sense. Thanks for the suggestion! Done.

**Review by diogomatsubara — APPROVED** [review]
Thanks!

**Pull request merged**